django_recaptcha/static/django_recaptcha/js/widget_v3.js

```diff
@@ -0,0 +1,52 @@
+(function () {
+
+  function main() {
+    grecaptcha.ready(function () {
+      for (const captchaElement of document.querySelectorAll(".g-recaptcha")) {
+
+        const widgetUuid = captchaElement.getAttribute("data-widget-uuid");
+        if (!widgetUuid) {
+          console.warn("reCAPTCHA widget with missing UUID");
+          break;
+        }
+        console.log(`found reCAPTCHA widget with UUID '${widgetUuid}'`);
+
+        const formElement = captchaElement.closest("form");
+        if (!formElement) {
+          console.warn(`reCAPTCHA widget with UUID '${widgetUuid}' is not part of a form`);
+          break;
+        }
+
+        const publicKey = captchaElement.getAttribute("data-sitekey");
+        if (publicKey === null) {
+          console.warn(`public key missing missing from reCAPTCHA widget with UUID '${widgetUuid}'`);
+          break;
+        }
+
+        // Google reCAPTCHA expects an object with an optional single key-value
+        // pair to communicate the name of the action.
+        const actionName = captchaElement.getAttribute("data-action");
+        const config = actionName !== null ? { action: actionName } : {};
```

Comment:
**Nitpick:** @alanverresen I don't think this rewritten code makes things much clearer. What about this?

 ```suggestion
         const actionName = captchaElement.getAttribute("data-action");
         const config = {};
         if (actionName !== null) {
           config.action = actionName;
         }
```

Reply:
I'm fine with both, so if you think yours is clearer, we'll go with that.